docs/how-to-guides/create-a-user-defined-function.md

```diff
@@ -0,0 +1,913 @@
+Overview    : Maintains a rolling sum of the last 3 integers of a stream.
+Type        : AGGREGATE
+Jar         : /opt/ksqldb-udfs/example-udfs-0.0.1.jar
+Variations  : 
+
+	Variation   : ROLLING_SUM(val INT)
+	Returns     : INT
+	Description : Sums the previous 3 integers of a stream, discarding the oldest elements as new ones arrive.
+```
+
+Create a stream named `s3`:
+
+```sql
+CREATE STREAM s3 (
+    a VARCHAR KEY,
+    b INT
+) WITH (
+    kafka_topic = 's3',
+    partitions = 1,
+    value_format = 'avro'
+);
+```
+
+Insert some rows into the stream:
+
+```sql
+INSERT INTO s3 (a, b) VALUES ('k1', 3);
+INSERT INTO s3 (a, b) VALUES ('k1', 5);
+INSERT INTO s3 (a, b) VALUES ('k1', 7);
+INSERT INTO s3 (a, b) VALUES ('k2', 6);
+INSERT INTO s3 (a, b) VALUES ('k2', 2);
+```
+
+Execute a push query. Recall what `rolling_sum` does. It aggregates the previous three elements together, sums them up, and emits their output.
+
+```sql
+SELECT a, rolling_sum(b) AS MOVING_SUM FROM s3 GROUP BY a EMIT CHANGES;
+```
+
+Your output should look like the following. `k1` sums `3`, `5`, and `7` together to get a result of `15`. `k2` sums `6` and `2` together to get a result of `8`.
+
+```
++--------------------------------------------------------------+--------------------------------------------------------------+
+|A                                                             |MOVING_SUM                                                    |
++--------------------------------------------------------------+--------------------------------------------------------------+
+|k1                                                            |15                                                            |
+|k2                                                            |8                                                             |
+```
+
+Insert some more rows and shift older elements out of the aggregate:
+
+```sql
+INSERT INTO s3 (a, b) VALUES ('k1', 9);
+INSERT INTO s3 (a, b) VALUES ('k2', 1);
+INSERT INTO s3 (a, b) VALUES ('k2', 6);
+```
+
+Run the query again:
+
+```sql
+SELECT a, rolling_sum(b) AS MOVING_SUM FROM s3 GROUP BY a EMIT CHANGES;
+```
+
+And you should now see these results. In `k1`, the previous three values are now `5`, `7`, and `9`. In `k2`, the elements are `2`, `1`, and `6`.
+
+```
++--------------------------------------------------------------+--------------------------------------------------------------+
+|A                                                             |MOVING_SUM                                                    |
++--------------------------------------------------------------+--------------------------------------------------------------+
+|k1                                                            |21                                                            |
+|k2                                                            |9                                                             |
+```
+
+## Working with structs
```

Comment:
we can probably omit this for now, or just add a placeholder, but this guide might want to consider covering `@UdfSchemaProvider` as well (and it can just point to examples in the existing code base, like `Abs`

Reply:
I punked out and just ended up mentioning it. I feel a little bad since it's an important topic. Can you think of a graceful way to introduce it without making the tutorial much more lengthy?